mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -1146,6 +1202,12 @@ ParseResult CallOp::parse(OpAsmParser &parser, OperationState &result) {
   TypeAttr calleeType;
   SmallVector<OpAsmParser::UnresolvedOperand> operands;
 
+  // Default to C Calling Convention if no keyword is provided.
+  result.addAttribute(
+      getCConvAttrName(result.name),
+      CConvAttr::get(parser.getContext(), parseOptionalLLVMKeyword<CConv>(
+                                              parser, result, LLVM::CConv::C)));
+
   // Parse a function pointer for indirect calls.
```

Comment:
The printer seems to print first the function type and then the CConvAttr, but the parser here seems reverse. Is there a test for this?

Reply:
There is a roundtrip test in `func.mlir` that seems to cover this for direct function calls. It might be a good idea to add a small test for indirect calls, though.

Reply:
If I am not completely confused, which may be the case it is still early :), then the vararg function type is printed on line 1125? The code before printing the calling convention is just accessing the type of the callee to figure out if the call is a var arg call. 

I agree that it makes sense to add a test!